src/_posts/2019-12-30-backstopjs-tutorial.md

```diff
@@ -0,0 +1,257 @@
+
+* `cookiePath`: This enables you to set different cookies for different scenarios
+* `url`: This is the full address of the web page being tested. If you're using a Docker container to host your site you may use the name of the container, like `http://website:8080/myPage`. Otherwise, you may run it locally and use something like `http://localhost:8080/myPage`.
+* `readyEvent`: Listen out for a console log telling you the page is fully loaded before starting. This is useful for repeatability.
+* `readySelector`: Similar to the above, this configures Backstop to wait until a particular element (defined by CSS selector) is appearing before starting. **I would strongly recommend using this setting and setting it to something that won't appear on any of your error pages**. If your service doesn't work during a visual regression test you may not know until after you get the report and are staring at a diff between your reference and a 404 page. But if your `readySelector` fails you get a timeout error in the output that lets you know that the expected page hasn't loaded, so you get the feedback sooner.
+* `delay`: **I strongly recommend that you never use this setting if you can avoid it whatsoever**. It allows you to set an arbitrary time to wait for the page to load before assuming it'll be ready to test. By default it is 0, which means no delay. If you find yourself using this setting, it's because you haven't found a reliable method to tell the browser that the page is loaded. You should only use this as an absolute last resort.
+* `hideSelectors`/`removeSelectors`: If you have some problematic element on the page that you either can't rely on to load in a reliable, timely fashion, or which has some random element that changes each time it's loaded then you can hide those elements using either of these properties.
+* `scrollToSelector`: BackstopJS will capture the entire document, not just the visible section (unless configured otherwise). However, you may want to trigger some event via scrolling. This setting makes Backstop scroll to a particular selector.
+* `selectors`: By default BackstopJS will capture the entire document. But if you want to test a specific set of regions, you can use this to limit the elements used to generate the snapshot. It's the opposite of `hideSelectors`/`removeSelectors` (but they can be used together).
```

Comment:
This is very useful if you have a complex page and want to break it down into several smaller tests. This adds to the run time but makes identifying specific regressions much easier.

Reply:
Good point, I'll add it in 👍

Reply:
Added in e1d961aa